src/local_pathfinding/local_pathfinding/obstacles.py

```diff
@@ -101,7 +100,132 @@ def update_reference_point(self, reference: HelperLatLon):
+        next_waypoint (HelperLatLon): Lat/lon position of the next global waypoint.
+        all_land_data (MultiPolygon): MultiPolygon of absolutely all land polygons known to
+                                      Sailbot.
+        bbox_buffer_amount (float): The amount of square buffer around Sailbot and around the next
+                                    global waypoint. In degrees lat/lon.
+    """
+
+    def __init__(
+        self,
+        reference: HelperLatLon,
+        sailbot_position: HelperLatLon,
+        next_waypoint: HelperLatLon,
+        all_land_data: MultiPolygon,
+        bbox_buffer_amount: float,
+    ):
+        super().__init__(reference, sailbot_position)
+        self.next_waypoint = next_waypoint
+        self.all_land_data = all_land_data
+        self.bbox_buffer_amount = bbox_buffer_amount
+        self._update_land_collision_zone()
+
+    def _update_land_collision_zone(
+        self, state_space: Polygon = None, land_multi_polygon: MultiPolygon = None
+    ) -> None:
+        """
+        Updates the Land object's collision zone with a MultiPolygon representing
+        all land obstacles within either a state_space input as an argument
+        or a default rectangle that bounds boxes around Sailbot and the next global waypoint.
+        """
+        if land_multi_polygon is not None:  # for testing
+
+            collision_zone = land_multi_polygon.buffer(0)
+
+            if collision_zone.geom_type == "Polygon":
+                collision_zone = MultiPolygon([collision_zone])
+
+            self.collision_zone = collision_zone
+            return
+
+        if state_space is None:  # create a deafult one
+
+            sailbot_box = Point(
+                self.sailbot_position_latlon.longitude, self.sailbot_position_latlon.latitude
+            ).buffer(self.bbox_buffer_amount, cap_style=3, join_style=2)
+
+            waypoint_box = Point(self.next_waypoint.longitude, self.next_waypoint.latitude).buffer(
+                self.bbox_buffer_amount, cap_style=3, join_style=2
+            )
+            state_space = box(*MultiPolygon([sailbot_box, waypoint_box]).bounds)
+
+        latlon_polygons = self.all_land_data.intersection(state_space)
+
+        xy_polygons = Land._latlon_polygons_to_xy_polygons([latlon_polygons], self.reference)
+
+        collision_zone = MultiPolygon(xy_polygons)
+
+        if len(collision_zone.geoms) > 0:
+            # if collision_zone is empty, buffer(0) will return an empty Polygon
+            # buffer(0) will repair invalid/overlapping geometry
+            # otherwise we cant run .contains() on it
+            collision_zone = collision_zone.buffer(0)
+
+            if collision_zone.geom_type == "Polygon":
+                collision_zone = MultiPolygon([collision_zone])
+        else:
+            collision_zone = MultiPolygon()
+
+        self.collision_zone = collision_zone
+
+    @staticmethod
+    def _latlon_polygons_to_xy_polygons(
```

Comment:
Could change the name to `_latlon_multipolygon_to_xy_polygons` as it's similar to the `_latlon_polygon_to_xy_polygon`.

Reply:
I named it `_latlon_polygon_list_to_xy_polygon_list` as the function does not expect a MultiPolygon parameter, rather just a list of polygons. What do you think?

Reply:
If the alternative to having nested functions makes it more unreadable, then it's fine to keep it as it is.